### ISSUE TITLE
Add a hook into `getIsCacheableRequest()`

### ISSUE THREAD

**Issue by benface** [putyourlightson/craft-blitz#306]
### Is your feature request related to a problem? Please describe.
Most pages on my site have user-specific content when the user is logged in, but most requests are from anonymous users who should all see the same content, so I would like to optimize for the latter as a "quick performance win" (as opposed to refactoring a large part of my code with `craft.blitz.getUri()` or `craft.blitz.getTemplate()`). So basically, I would like to only render the cached version of a page if the user is not logged in. I understand that is a specific requirement, so I don't expect a setting to do just that, *but* it would be very nice to be able to control precisely whether a request should use the cache or not.

### Describe the solution you would like
It looks like whether a request should use the cache or not is determined by the `getIsCacheableRequest()` method of the `CacheRequestService`. So I think if that method triggered an event, a custom plugin/module could listen to that event and set a property on the `$event` object to prevent using the cache for the current request, if it wanted to. In my case, I imagine it would look something like this:
```php
Event::on(CacheRequestService::class,
    CacheRequestService::EVENT_GET_IS_CACHEABLE_REQUEST,
    function(GetIsCacheableRequestEvent $event) {
        if (Craft::$app->getUser()->getIdentity() !== null) {
            $event->isCacheable = false;
        }
    }
);
```

### Describe alternatives you have considered
As mentioned above, I know the ideal solution would be to use the cache for every request and inject dynamic content with `craft.blitz.getUri()` or `craft.blitz.getTemplate()`, but in my case, that represents a lot of work for very little gain.

Thank you for your consideration.

**Comment by bencroker** [comment]
You can use the `EVENT_BEFORE_GET_RESPONSE` event, setting `isValid` to `false` to prevent a cached version being output.
https://github.com/putyourlightson/craft-blitz/blob/de6d2de6d5f119af33ccc7e708cc7861ca13fe5f/src/services/CacheRequestService.php#L222

**Comment by benface** [comment]
Oh, awesome! I did check the documentation and the GitHub issues beforehand, but I should've dug a little deeper in the code probably. :) Thanks!

**Comment by benface** [comment]
Hmm, there is a problem with that solution, @bencroker. While setting `isValid` to `false` indeed prevents a cached version of the current page from being used, it doesn't seem to prevent *writing* to the cache. It recaches the page every time, which is obviously a problem because then an anonymous user would see a random user's version of the page.

**Comment by bencroker** [comment]
You're right Ben. I just added an `isCacheableRequest` event in https://github.com/putyourlightson/craft-blitz/commit/aea25aea5eac5cbfa5a8d723769df39778304438, on which you can set `isValid` to `false` to cancel the check sooner.

**Comment by benface** [comment]
You're on fire today! Thank you so much.

**Comment by bencroker** [comment]
You caught me at a good time!

**Comment by bencroker** [comment]
Released in version 3.9.0.
